Use a secret config file to auto-populate fields

jia1 opened the pull request:
Signed-off-by: Jia Yee <jiayeerawr@gmail.com>

Fixes #17

jia1 committed:
Add example config file to auto-populate fields

Signed-off-by: Jia Yee <jiayeerawr@gmail.com>

jia1 committed:
Auto-populate fields from config file (incomplete)

Incomplete because form value fields refuse to interpolate the locals,
the interpolation works outside of the value fields.

Signed-off-by: Jia Yee <jiayeerawr@gmail.com>

thenaesh commented on src/html/entrypoint.pug:
```diff
@@ -54,57 +54,57 @@ html
                         label.control-label.col-sm-2(for='send_bcc') BCC:
                         .col-sm-10
-                            input#send_bcc.form-control(type='email', value='marketing@nuscomputing.com')
+                            input#send_bcc.form-control(type='email', value='#{config.email_header.bcc}')
```
```javascript
value=`${config.email_header.bcc}`
```

Use ES6 interpolation syntax here. I don't think single quotes does any interpolation (at least I don't see it in the Pug docs).

jia1 replied:
ES6 interpolation works! I guess I was reading out-of-date information for Pug.

jia1 committed:
Fix field auto-population

Interpolation in Pug is no longer supported. Do interpolation with ES6
instead.

Signed-off-by: Jia Yee <jiayeerawr@gmail.com>

jia1 committed:
Fix recipient email auto-population

Originally a wrong reference to sender instead (i.e.
.sender_email instead of .recipient_email)

Signed-off-by: Jia Yee <jiayeerawr@gmail.com>

jia1 committed:
Fix value for default upload source path

Replace '..' with 'src' because navigation begins from project root

Signed-off-by: Jia Yee <jiayeerawr@gmail.com>

jia1 commented:
Now awaiting review. The application side is cleared, but there are remaining issues in the server side.

E.g. File/folder permissions, web server configuration to serve new folders
